### PR TITLE
fix: two high-risk vulnerabilities

### PR DESCRIPTION
**Pull request by StarsRivers**
fix:Incorrect conversion between integer types
fix:Use of insecure HostKeyCallback implementation

### CODE REVIEW

**Comment by StarsRivers** [comment]
两个漏洞均由CodeQL扫描发现，本人进行修复。
https://github.com/StarsRivers/alist/security/code-scanning/5
https://github.com/StarsRivers/alist/security/code-scanning/13

File: drivers/sftp/util.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func (d *SFTP) initClient() error {
 	config := &ssh.ClientConfig{
 		User:            d.Username,
 		Auth:            []ssh.AuthMethod{auth},
-		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		HostKeyCallback: ssh.FixedHostKey(hostKey),
```
Comment (xhofe) [inline drivers/sftp/util.go:26]: where is `hostKey`?

File: server/handles/ssologin.go [hunk 2]
```diff
@@ -132,7 +132,7 @@ func autoRegister(username, userID string, err error) (*model.User, error) {
 		ID:         0,
 		Username:   username,
 		Password:   random.String(16),
-		Permission: int32(setting.GetInt(conf.SSODefaultPermission, 0)),
+		Permission: int32(strconv.ParseInt(setting.GetInt(conf.SSODefaultPermission, 0)), 10, 32),
```
Comment (xhofe) [inline server/handles/ssologin.go:135]: There are too many arguments in conversion to int32, and can't pass a int value to `ParseInt`

**Review by xhofe — COMMENTED** [review]
I'm sorry that we can't accept this PR, thanks for your contribution!

**Pull request closed without merging**